### ISSUE TITLE
dtypes lost in to_pandas conversion 

### ISSUE THREAD

**Issue by yifan** [elastic/eland#274]
I did .to_pandas on a eland DataFrame. This is the dtypes for original eland DataFrame:

.  | .
-- | --
publisher | object
authors |object
pub_data | datetime64[ns]
score | float64

And the result pandas DataFrame dtypes:

.  | .
-- | --
publisher | object
authors |object
pub_data | object
score | float64

The pub_data datetime type is lost. Is this expected behaviour?

**Comment by sethmlarson** [comment]
We should be maintaining types. Were there any entries in the Pandas dataframe that were not a datetime?

**Comment by yifan** [comment]
@sethmlarson I have updated the issue with additional entry score. Its type is float64, and it is  correct  after conversion.

**Comment by sethmlarson** [comment]
@yifan In the `pub_data` Series were all objects datetimes or are there any items that were serialized as non-datetimes?

Where `pd_df` is the Pandas DataFrame you receive after calling `.to_pandas()`:

```python
from datetime import datetime

print(pd_df.pub_data[pd_df.pub_data.apply(lambda item: not isinstance(item, datetime))])
```

Is the resulting series empty or what items are in that series?

**Comment by yifan** [comment]
hi @sethmlarson the result series in empty when I run above code. Here is an example entry for the datetime series in pd_df:
`2020-04-02 11:10:00+00:00`. Do you think it could be a format issue? (having space instead of 'T')

**Comment by sethmlarson** [comment]
@yifan Okay, I'll investigate this, could you also give me your Python, Elasticsearch, and Eland versions?

**Comment by yifan** [comment]
Thank you, @sethmlarson. Here is the versions of my environment:
```
Python 3.8.5
Elasticsearch 7.4.0
eland==7.9.0a1
```

**Comment by yifan** [comment]
I am not sure if this `es_info()` can help or not, here it is:
```
Index:
 es_index_field: _id
 is_source_field: False
Mappings:
 capabilities:
            es_field_name  is_source es_dtype es_date_format        pd_dtype  is_searchable  is_aggregatable  is_scripted aggregatable_es_field_name
authors           authors       True     text           None          object           True            False        False                       None
key                   key       True  keyword           None          object           True             True        False                        key
pub_date         pub_date       True     date           None  datetime64[ns]           True             True        False                   pub_date
publisher       publisher       True  keyword           None          object           True             True        False                  publisher
section           section       True  keyword           None          object           True             True        False                    section
subjects         subjects       True     text           None          object           True            False        False           subjects.keyword
tags                 tags       True     text           None          object           True            False        False               tags.keyword
title               title       True     text           None          object           True            False        False                       None
```

**Comment by Fju** [comment]
> hi @sethmlarson the result series in empty when I run above code. Here is an example entry for the datetime series in pd_df:
> `2020-04-02 11:10:00+00:00`. Do you think it could be a format issue? (having space instead of 'T')

It seems like this is the problem. In ([eland/common.py, line 87](https://github.com/elastic/eland/blob/0dd247b69364951b8402a00dd8463a6a69bdf671/eland/common.py#L87)) you can see the logic for converting the ElasticSearch datetime string (or Integer / Float)  to a pandas' datetime type. As you can see eland expects the T separator, so converting the string `"2020-04-02 11:10:00+00:00"` (without the T separator) won't work. The eland implementation matches the datetime formats of ElasticSearch. If you want to look it up in the documentation, here is a [link](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-date-format.html#built-in-date-formats).

In short, the data you query is in a wrong format. You should check how the data is written to ElasticSearch. As far as I can tell, there is no error in eland since it follows the date formats specified in the documentation.

Cheers :v:

**Comment by sethmlarson** [comment]
Could you try this again in the latest Eland? This section of the code has changed and potentially fixed via another enhancement?

**Comment by sethmlarson** [comment]
Haven't heard anything new on this issue in some time, going to close it.
